app.js

```diff
@@ -34,26 +34,25 @@ class NetworkSpeedCheck {
     });
   }
 
-  checkUploadSpeed(options) {
-    const dataSizeInKb = 20;
+  checkUploadSpeed(options, fileSizeInBytes = 2000000) {
     let startTime;
-    const data = '{"data": "' + this.generateTestData(20) + '"}';
+    const data = '{"data": "' + this.generateTestData(fileSizeInBytes / 1000) + '"}';
     return new Promise((resolve, _) => {
       var req = http.request(options, res => {
-        res.setEncoding('utf8');
-        res.on('data', () => {
-          startTime = new Date().getTime();
```

Comment:
If `startTime` isn't set when the data transfer starts, startTime will always be null